### DISCUSSION TITLE
MQTT only shows cluster data

### DISCUSSION THREAD

**Discussion by Nico1320** [mKeRix/room-assistant#850] [Q&A] [answered]
I just did a fresh install as HA addon and my devices just don't show up on Mosquito. Is the addon broken on HASS OS 6.2?

> global:
  instanceName: Master Bedroom
  integrations:
    - homeAssistant
    - bluetoothLowEnergy
bluetoothLowEnergy:
  allowlist:
    - 48a01dbe-8f28-40ed-bc7c-0388a0ced39f-1-2
  maxDistance: 10
  onlyIBeacon: true
  updateFrequency: 5
  tagOverrides:
    48a01dbe-8f28-40ed-bc7c-0388a0ced39f-1-2:
      name: S8
      measuredPower: -65

**Comment by k41zen** [comment]
I have exactly the same issue. Had 6 x Pi's running RA working with my old HA instance on my NAS. All fine. My HA broke so I did a fresh install. Now the PI's show in MQTT Broker but no devices. In MQTT Explorer it sees all Pi's and ble devices too. Just not getting any devices created or showing in HA

**Comment by mKeRix** [comment, marked as answer]
If it's just not showing up in HA: can you confirm that the MQTT integration was configured with auto discovery on? In the HA config flow it should be a toggle to switch during setup, for YAML config it's just a property to edit. You may have to re-add the integration to change this option.

Of course, if you don't see it on MQTT at all (or you already have other working, auto-discovered MQTT entities) this is likely not the cause.

**Reply by Nico1320** [reply]
Auto-discovery is on, Even using MQTT explorer I can only see cluster-size/cluster-leader.

**Reply by mKeRix** [reply]
In that case it sounds like it's an issue with the BLE part rather than the HA one. What do your logs say? Do you see a bunch of "discovered device xyz" log messages - and more specifically, do you see one for your specific beacon ID? What is broadcasting the beacon, the HA companion app?

**Reply by Nico1320** [reply]
I'm using Beacon Simulator
Logs:
> debug - BluetoothService: Adapter 0 went into state poweredOn
error - BluetoothService: Failed to start advertising instance via BLE: Command Disallowed
info - BluetoothLowEnergyService: Discovered new BLE peripheral 48a01dbe8f2840edbc7c0388a0ced39f-1-2 with ID 48a01dbe8f2840edbc7c0388a0ced39f-1-2 and RSSI -83

**Reply by mKeRix** [reply]
Looks like the discovery is working ok then, can you simply try updating your config above and replacing the reference to the ID with `48a01dbe8f2840edbc7c0388a0ced39f-1-2`? (essentially this removes the dashes in the iBeacon UUID)

**Reply by Nico1320** [reply]
That did it. it works now

**Comment by k41zen** [comment]
I'm still having issues. RA in debug mode isn't seeing any of my devices which are a mix of companion app, Apple Watch, 2 x Nut mini key fobs. It does find every time my Ember mug sat on my desk though so I added that to the config but see nothing about adding it to HA nor does it show in MQTT Explorer. 

This is RA -v: 

![Screenshot 2021-08-06 at 11 13 26](https://user-images.githubusercontent.com/43755140/128495558-4a2a31f5-4719-447a-b32c-807ca0543e97.png)

MQTT Explorer:

![Screenshot 2021-08-06 at 11 12 09](https://user-images.githubusercontent.com/43755140/128495587-939cfe7d-ca74-4341-81b0-e62ebce2aa60.png)

**Reply by Nico1320** [reply]
To me, it sounds like its something to do with auto-discovery. Are you using Mosquito? Is discovery enabled? If the devices are showing up on MQTT explorer it likely means the issue lies in HA.

**Reply by k41zen** [reply]
Yes to Mosquito and yes to discovery is enabled, also tried explicitly defining this in the yaml. Totally agree its HA as the 6 PI's have been working since I set it up in February. When I spin up my old broken instance they update fine. I've copied and replicated everything I can see from the old HA instance to the new but still can only see the PI's in the Broker. Can't see any devices.

**Reply by k41zen** [reply]
How about that. Change the config on a PI to Bluetooth Classic and suddenly the devices seen (Apple Watch, Nut Mini's etc) are registered in the Broker and devices created in HA. Odd. So it "is" a BLE issue 🤔

**Reply by Nico1320** [reply]
Hmmm strange, were those devices in MQTT explorer retained messages?

**Reply by k41zen** [reply]
No, it was a new session.
